amd/src/mod_hsuforum_save_draft.js

```diff
@@ -0,0 +1,128 @@
+define([], function() {
+    /**
+     * Makes AJAX call to write current text for forum/discussion/post to DB for restoration later on
+     * @param forumid int
+     * @param discussionid int
+     * @param postid int
+     * @param text string
+     * @param userid int
+     */
+    function autoSave(forumid, discussionid, postid, text, userid) {
+        if ((forumid != '' || discussionid != '') && text != '' && userid != '') {
+            $.ajax({
+                url: "save_post_draft.php",
+                method: "POST",
+                data: {
+                    forumid: forumid,
+                    discussionid: discussionid,
+                    postid: postid,
+                    text: text,
+                    userid: userid
+                },
+                dataType: "text",
+                success: function() {
+                    // Do nothing, write-only call
+                }
+            });
+        }
+    }
+
+    /**
+     * Makes AJAX call to restore draft if it exists
+     * @param forumid
+     * @param discussionid
+     * @param postid
+     * @param userid
+     */
+    function fetchAutoSave(forumid, discussionid, postid, userid) {
+        if ((forumid != '' || discussionid != '' || postid != '') && userid != '') {
+            $.ajax({
+                url: "get_post_draft.php",
+                method: "GET",
+                data: {
+                    forumid: forumid,
+                    discussionid: discussionid,
+                    postid: postid,
+                    userid: userid
+                },
+                dataType: "text",
+                success: function(data) {
+                    window.drafttext = data;
+                }
+            });
+        }
+    }
+
+    return {
+        init: function(forumid, discussionid, userid) {
+            $(function() {
+                let timer;
+                // This is to restore the draft when the page loads and programatic click occurs
+                let pageloadclick = false;
+
+                // Bind click to the reply
+                $('.hsuforum-reply-link').on('click', function() {
+                    window.drafttext = '';
+
+                    let parent = $(this).closest('.hsuforum-post-wrapper');
+                    let postid = $(parent).attr('data-postid');
```

Comment:
Do you think its worth while validating these params?
`if (posted !== undefined) {
      fetchAutoSave(forumid, discussionid, postid, userid);
      ........
  }`

Reply:
Different sections will behave differently. That block is for a direct post reply, so it will always have that wrapper. I've done individual event handlers for each specific draft type. In this event that HTML and attr will always exist